pkg/networkmanager/utils.js

```diff
@@ -56,6 +56,35 @@ export function ip_metric_from_text(text) {
     throw cockpit.format(_("Invalid metric $0"), text);
 }
 
+export function ip_network_address(address, prefix_len) {
+    const addrCIDR = address.toString() + "/" + prefix_len;
+    const ip_obj = address.kind() === "ipv4" ? ipaddr.IPv4 : ipaddr.IPv6;
+
+    try {
+        return ip_obj.networkAddressFromCIDR(addrCIDR).toString();
+    } catch (_e) {
+        return null;
+    }
+}
+
+export function ip_first_usable_address(address, prefix) {
+    try {
+        const addr_cidr = address.toString() + "/" + prefix;
+
+        if (address.kind() === "ipv4") {
+            const netAddr = ipaddr.IPv4.networkAddressFromCIDR(addr_cidr);
+            netAddr.octets[3] += (prefix < 31) ? 1 : 0;
+            return netAddr.toString();
+        } else {
+            const netAddr = ipaddr.IPv6.networkAddressFromCIDR(addr_cidr);
+            netAddr.parts[7] += (prefix < 127) ? 1 : 0;
+            return netAddr.toString();
+        }
+    } catch (_e) {
+        return null;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22608-06508236-20251127-101123-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)